OCPBUG3527: Added "allow-from-router" part as well as references. (#57464)
openshift/openshift-docs: modules/nw-networkpolicy-about.adoc

`NetworkPolicy` objects are additive, which means you can combine multiple `NetworkPolicy` objects together to satisfy complex network requirements.

For example, for the `NetworkPolicy` objects defined in previous samples, you can define both `allow-same-namespace` and `allow-http-and-https` policies within the same project. Thus allowing the pods with the label `role=frontend`, to accept any connection allowed by each policy. That is, connections on any port from pods in the same namespace, and connections on ports `80` and `443` from pods in any namespace.

[id="nw-networkpolicy-allow-from-router_{_content-type}"]
== allow from router

> Allows router regardless of the configuration:

Reviewer: Allows traffic from router regardless of endpoint publishing strategy.

Reviewer: Can also add allow-from-hostnetwork policy
<pre>
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: allow-from-hostnetwork
spec:
  ingress:
  - from:
    - namespaceSelector:
        matchLabels:
          policy-group.network.openshift.io/host-network: ""
  podSelector: {}
  policyTypes:
  - Ingress
  </pre>

Author: As another part or step?

`NetworkPolicy` objects are additive, which means you can combine multiple `NetworkPolicy` objects together to satisfy complex network requirements.

For example, for the `NetworkPolicy` objects defined in previous samples, you can define both `allow-same-namespace` and `allow-http-and-https` policies within the same project. Thus allowing the pods with the label `role=frontend`, to accept any connection allowed by each policy. That is, connections on any port from pods in the same namespace, and connections on ports `80` and `443` from pods in any namespace.

[id="nw-networkpolicy-allow-from-router_{context}"]

> == allow from router

Reviewer 2: ```suggestion
== Using the allow-from-router network policy
```

Reviewer 2: This is a tough one because of the name of the policy. But I tried to make suggestions that keep the technical language in tact while limiting allow as a verb: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#a

Author: Yes. Was puzzled whether to keep it as it is or to change the heading. Since it was already used in the previous sections. I followed the same.

> Allows traffic from router regardless of the router configuration:

Reviewer 2: ```suggestion
Use the following `NetworkPolicy` to allow external traffic regardless of the router configuration:
```

To make a project allow external traffic regardless of the configuration, add the following `NetworkPolicy` object.

> `policy-group.network.openshift.io/ingress` label supports both Openshift-SDN and OVN-Kubernetes. 

Reviewer 2: I think this probably needs to be a <1> and under the yaml block.

Author: Done.

[source,yaml]
----
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: allow-from-router
spec:
  ingress:
  - from:
    - namespaceSelector:
        matchLabels:

>           policy-group.network.openshift.io/ingress:""

Reviewer 2: Like here is where I would put it and then write <1> Label supports both OpenShift-SDN and OVN-K8s

>           policy-group.network.openshift.io/ingress:""

Author: Done.

  podSelector: {}
  policyTypes:
  - Ingress
----

[id="nw-networkpolicy-allow-from-hostnetwork_{context}"]

> == allow from hostnetwork

Reviewer 2: ```suggestion
== Using the allow-from-hostnetwork network policy
```

Author: Done

> Allows traffic from host network:

Reviewer 2: ```suggestion
Add the following `allow-from-hostnetwork` `NetworkPolicy` object to direct traffic from the host network pods:
```

Author: Done.

> To make a project allow external traffic from the hostnetwork pods, add the following `NetworkPolicy` object.

Reviewer 2: ```suggestion
```

Author: Done.
